### PR TITLE
Update to the JupyterLab 3.0 prebuilt extension system

### LINKED ISSUE #10: Add support for prebuilt extension

### PR DESCRIPTION
**Pull request by jtpio**
Fixes #10 

Using the upgrade script to do most of the lifting.

image

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        python -m pip install .
+
+        jupyter labextension list 2>&1 | grep -ie "@timkpaine/jupyterlab_miami_nights.*OK"
+        python -m jupyterlab.browser_check
+
+        check-manifest -v
+
+        pip install build
+        python -m build --sdist
+        cp dist/*.tar.gz myextension.tar.gz
```
Comment (timkpaine) [inline .github/workflows/build.yml:63]: should this be renamed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        pip uninstall -y myextension jupyterlab
```
Comment (timkpaine) [inline .github/workflows/build.yml:64]: same

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        rm -rf myextension
```
Comment (timkpaine) [inline .github/workflows/build.yml:65]: same

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: myextension-sdist
```
Comment (timkpaine) [inline .github/workflows/build.yml:69]: same

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        path: myextension.tar.gz
```
Comment (timkpaine) [inline .github/workflows/build.yml:70]: same

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    - uses: actions/download-artifact@v2
+      with:
+        name: myextension-sdist
+    - name: Install and Test
+      run: |
+        set -eux
+        # Remove NodeJS, twice to take care of system and locally installed node versions.
+        sudo rm -rf $(which node)
+        sudo rm -rf $(which node)
+        pip install myextension.tar.gz
```
Comment (timkpaine) [inline .github/workflows/build.yml:93]: same

**Review by timkpaine — CHANGES_REQUESTED** [review]

**Comment by timkpaine** [comment]
@jtpio this is up on pypi, mind testing to make sure it works as expected?

**Comment by jtpio** [comment]
Thanks @timkpaine :+1: 

Looks good when installing on Binder:

image

**Comment by jtpio** [comment]
Although some styling persists after switching back to the light theme:

image

**Comment by jasongrout** [comment]
> Although some styling persists after switching back to the light theme:

To fix this, likely you just need to install the `style` and `styleModule` keys in the package.json:
https://github.com/timkpaine/jupyterlab_miami_nights/blob/c5ce7911967dd3f30f25d916237381c73f6843b9/package.json#L32
https://github.com/timkpaine/jupyterlab_miami_nights/blob/c5ce7911967dd3f30f25d916237381c73f6843b9/package.json#L96

See https://github.com/telamonian/theme-darcula/pull/33 for a similar change for another theme.

**Pull request merged**